src/mode/astro_highlight_rules.js

```diff
@@ -2,82 +2,104 @@
 
 var AstroHighlightRules = function () {
   HtmlHighlightRules.call(this);
 
   var astro = {
-    token: "paren.quasi.start.astro.level3",
+    token: "paren.quasi.start",
     regex: /{/,
-    next: "inline-js-start"
+    next: function (state, stack) {
+      if (state !== "start") {
+        if (state.indexOf("attribute-equals") !== -1) {
+          stack.splice(0);
+          stack.unshift("tag_stuff");
+        } else {
+          stack.unshift(state);
+        }
+      }
+      return "inline-js-start";
+    }
   };
 
   for (var key in this.$rules) {
-    if (key.startsWith("js") || key.startsWith("css") || key.startsWith("comment")) continue;
+    if (
+      key.startsWith("js") ||
+      key.startsWith("css") ||
+      key.startsWith("comment")
+    )
+      continue;
     this.$rules[key].unshift(astro);
   }
 
   this.$rules.start.unshift({
     token: "comment",
-    regex: "---",
+    regex: /^---$/,
     onMatch: function (value, state, stack) {
       stack.splice(0);
       return this.token;
     },
-    next: "js-start"
+    next: "javascript-start"
   });
 
-  this.embedRules(JavascriptHighlightRules, "js-", [{
-    regex: "---",
-    token: "comment",
-    next: "start",
-    onMatch: function (value, state, stack) {
-      stack.splice(0);
-      return this.token;
+  this.embedRules(JavascriptHighlightRules, "javascript-", [
+    {
+      regex: /^---$/,
+      token: "comment",
+      next: "start",
+      onMatch: function (value, state, stack) {
+        stack.splice(0);
+        return this.token;
+      }
     }
-  }]);
+  ]);
 
-  this.embedRules(JavascriptHighlightRules, "inline-js-", [{
-    regex: /}/,
-    token: "paren.quasi.end.astro.level3",
-    onMatch: function (value, state, stack) {
-      if (stack.length) {
-        if (stack.includes("inline-js-start")) {
+  this.embedRules(JavascriptHighlightRules, "inline-js-");
+
+  var astroRules = [
+    {
+      regex: /}/,
+      token: "paren.quasi.end",
+      onMatch: function (value, state, stack) {
+        if (stack[0] === "inline-js-start") {
           stack.shift();
           this.next = stack.shift();
-          if (this.next.indexOf("string") !== -1) return "paren.quasi.end";
+          if (this.next.indexOf("string") !== -1) 
+            return "paren.quasi.end";
           return "paren.rparen";
         } else {
-          if (stack.includes("string.attribute-value.xml0")) {
-            this.next = "string.attribute-value.xml0";
-          }
-          else if (stack.includes("tag_stuff")) {
-            this.next = "tag_stuff";
-          }
+          this.next = stack.shift() || "start";
           return this.token;
         }
-      } else {
-        this.next = this.nextState;
-        return this.token;
       }
     },
-    nextState: "start"
-  }, {
-    regex: /{/,
-    token: "paren.lparen",
-    push: "inline-js-start"
-  }]);
+    {
+      regex: /{/,
+      token: "paren.rparen",
```

Comment:
The regex here looks like this should be a left parenthesis?

Reply:
opps . the token should be  paren.lparen